test/test_autograd.py

```diff
@@ -101,6 +101,29 @@ def _function_test(self, cls):
 
         return x, y
 
+    def test_bwd_pre_hook(self):
```

Comment:
I think there should be another test that verifies returning a new gradient from a pre-backwards hook